File: LeetCode/1578.Minimum-Time-to-Make-Rope-Colorful.cpp
// 1578. Minimum Time to Make Rope Colorful

class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int n = colors.length();
        int ans = 0;
        for(int i=0; i<n; i++) {
            int maxi = 0;
            while(i<n-1 && colors[i] == colors[i+1]) {
                ans += neededTime[i];
                maxi = max(maxi , neededTime[i]);
                i++;
            }
            ans += neededTime[i];
            maxi = max(maxi , neededTime[i]);
            ans -= maxi;
        }
        return ans;
    }
};

File: LeetCode/1578-Minimum-Time-to-Make-Rope-Colorful.java
// 1578. Minimum Time to Make Rope Colorful

class Solution {
    public int minCost(String colors, int[] neededTime) {
        int n = neededTime.length;
        int l = 0, r = 0, ans = 0;
        while (r < n) {
            int mx = Integer.MIN_VALUE;
            while (r < n && colors.charAt(l) == colors.charAt(r)) {
                ans += neededTime[r];
                mx = Math.max(mx, neededTime[r]);
                r++;
            }
            ans -= mx;
            l = r;
        }
        return ans;
    }
}
